ocaml/testsuite/tests/typing-implicit-source-positions/commuting.ml

```diff
@@ -0,0 +1,52 @@
+(* TEST
+   * expect
+*)
+
+let f = fun ~(a:[%src_pos]) ~(b:[%src_pos]) () -> ()
+[%%expect{|
+val f : a:[%src_pos] -> b:[%src_pos] -> unit -> unit = <fun>
+|}]
+
+let _ = f ~b:Lexing.dummy_pos ~a:Lexing.dummy_pos () ;;
```

Comment:
The general idea here is good, but how do we know that the commuting is actually working? That is, this type-checks, but does it run correctly? We can't be sure from this test.